ipn/ipnstate/ipnstate.go

```diff
@@ -183,6 +185,26 @@ type PeerStatusLite struct {
 	NodeKey key.NodePublic
 }
 
+// UserID is a tailcfg.UserID that is JSON encoded as a string.
+type UserID tailcfg.UserID
+
+// MarshalJSON marshals a UserID as a JSON string.
+func (u UserID) MarshalJSON() ([]byte, error) {
+	return fmt.Appendf(nil, `"%d"`, u), nil
+}
+
+// UnmarshalJSON unmarshals a UserID from a JSON string.
+func (u *UserID) UnmarshalJSON(b []byte) error {
+	b = bytes.Trim(b, `"`)
```

Comment:
Unlikely to occur in practice, but this would successfully unmarshal `"12` and `""456""`. (We use it inside a bigger object so we get the outside validity check so it isn't a huge risk)

Reply:
true.  Unlikely, but I've gone ahead and updated to verify JSON string syntax.